File: kobiton-rest-api/get-devices-info/python/check-device-status.py
import os
import requests
import sys
import json
headers = {
'Accept': 'application/json'
}

username = os.getenv('KOBITON_USERNAME')
apiKey = os.getenv('KOBITON_API_KEY')

if username==None or apiKey==None:
  print("KOBITON_USERNAME and KOBITON_API_KEY variables are need to execute the script")
  sys.exit()

r = requests.get('https://api.kobiton.com/v1/devices?isOnline=true&isBooked=false', auth=(username, apiKey),
headers = headers)

if r.status_code != 200:
  print(r.status_code)
  print(r.text)
  sys.exit()

deviceList = json.loads(r.text)
cloudDevices = deviceList["cloudDevices"]

if len(cloudDevices) == 0:
  print("There is no online device")
  sys.exit()

print("Online device list:")
for device in cloudDevices:
  item = {
    "id": device["id"],
    "udid": device["udid"],
    "isBooked": device["isBooked"],
    "modelName": device["modelName"],
    "deviceName": device["deviceName"],
    "platformName": device["platformName"],
    "platformVersion": device["platformVersion"]
  }
  print(json.dumps(item, indent=4, sort_keys=True))

File: kobiton-rest-api/get-devices-info/nodejs/check-device-status.js
const request = require('request')
const BPromise = require('bluebird')

const sendRequest = BPromise.promisify(request)

const username = process.env.KOBITON_USERNAME
const apiKey = process.env.KOBITON_API_KEY

const headers = {
    'Accept': 'application/json'
}

async function getOnlineDevice() {
  let response
  try {
    response = await sendRequest({
      url: `https://api.kobiton.com/v1/devices?isOnline=true&isBooked=false`,
      json: true,
      method: 'GET',
      auth: {
          user: username,
          pass: apiKey
      },
      headers
      })
  }
  catch (err) {
    console.error('Error occured while sending request to Kobiton via API', err)
  }

  if (response.statusCode != 200) {
    console.log('status:', response.statusCode)
    console.log('body:', response.body)
    process.exit(1)
  }
  
  const {cloudDevices} = response.body
  const deviceList = await filterDeviceInfo(cloudDevices)

  if (!deviceList) {
    console.log('Online device list: ', deviceList)
  }
  else {
    console.log('There is no online device')
  }
}

async function filterDeviceInfo(deviceList) {
  if (deviceList.length === 0) {
    return null
  }

  const filteredDevices = deviceList.map((device) => {
    const {id, udid, isBooked, modelName, deviceName, platformName, platformVersion} = device
    return {id, udid, isBooked, modelName, deviceName, platformName, platformVersion}
  })

  return filteredDevices
}

async function main() {
  if(!username || !apiKey)
  {
      console.log('KOBITON_USERNAME and KOBITON_APIKEY variables are need to execute the script')
      process.exit(1)
  }

  await getOnlineDevice()
}

main()
